framework/src/modules/token/cc_commands/cc_transfer.ts

```
@@ -49,104 +46,154 @@ export class CCTransferCommand extends BaseCCCommand {
-		// 		if (ccm.fee >= MIN_RETURN_FEE * ctx.ccmSize && ccm.status === CCM_STATUS_OK) {
-		// 			await this._interopMethod.error(methodContext, ccm, CCM_STATUS_MIN_BALANCE_NOT_REACHED);
-		// 		}
-		// 		return;
-		// 	}
-		// 	receivedAmount -= minBalance;
-		// 	const [canonicalChainID, canonicalLocalID] = splitTokenID(canonicalTokenID);
-		// 	if (canonicalChainID.equals(CHAIN_ID_ALIAS_NATIVE)) {
-		// 		const supplyStore = this.stores.get(SupplyStore);
-		// 		const supply = await supplyStore.get(methodContext, canonicalLocalID);
-		// 		supply.totalSupply -= minBalance;
-		// 		await supplyStore.set(methodContext, canonicalLocalID, supply);
-		// 	}
-		// }
-
-		// if (tokenChainID.equals(ownChainID)) {
-		// 	const escrowStore = this.stores.get(EscrowStore);
-		// 	const escrowKey = Buffer.concat([ccm.sendingChainID, tokenLocalID]);
-		// 	const escrowData = await escrowStore.get(methodContext, escrowKey);
-
-		// 	escrowData.amount -= params.amount;
-		// 	await escrowStore.set(methodContext, escrowKey, escrowData);
-		// }
-
-		// const userStore = this.stores.get(UserStore);
-		// await userStore.addAvailableBalanceWithCreate(
-		// 	methodContext,
-		// 	recipientAddress,
-		// 	canonicalTokenID,
-		// 	receivedAmount,
-		// );
+		const userStore = this.stores.get(UserStore);
+		const recipientExist = await userStore.has(
+			methodContext,
+			userStore.getKey(recipientAddress, tokenID),
+		);
+
+		if (!recipientExist) {
+			if (fee < FEE_CCM_INIT_USER_STORE) {
+				throw new Error('Insufficient fee to initialize user account.');
+			} else {
+				// should add trsSender address
+				await this._tokenMethod.burn(methodContext, feeAddress, tokenID, amount);
+			}
+		}
+
+		if (tokenChainID.equals(ownChainID)) {
+			const escrowStore = this.stores.get(EscrowStore);
+			const escrowKey = escrowStore.getKey(sendingChainID, tokenID);
+			const escrowData = await escrowStore.get(methodContext, escrowKey);
+
+			if (escrowData.amount > amount) {
```

if there is not enough amount in escrow, it should throw an error